### DISCUSSION TITLE
Peer depenency issue with typescript 5.x project (and npm >= 7.x)

### DISCUSSION THREAD

**Discussion by brentropy** [tw-in-js/twind#473] [General]
When installing twind in my project I get peer dependency errors unless I add `--force` or `--legacy-peer-deps`.

```
npm ERR! While resolving: @twind/cdn@1.0.8
npm ERR! Found: typescript@5.0.4
npm ERR! node_modules/typescript
npm ERR!   peerOptional typescript@"^4.8.4" from @twind/core@1.1.3
npm ERR!   node_modules/@twind/cdn/node_modules/@twind/core
npm ERR!     @twind/core@"^1.1.3" from @twind/cdn@1.0.8
npm ERR!     node_modules/@twind/cdn
npm ERR!       dev @twind/cdn@"*" from the root project
```

I see that it is marked as an optional peer dependency (I didn't know that was a thing until now), but it isn't optional to have a compatible version if the project already has `typescript` as a dependency.

First of all, why does twind have typescript as a peer dependency? I am probably missing something, but I don't see how it gets used by the built package.

Second, if it is really needed is there any reason it couldn't specify something like `>=4.8.4` so it doesn't break when installing in projects that use newer versions? I am happy to submit a PR to do this if it makes sense.

I would like to use `twind` over `tailwindcss`, but I don't really want to revert back to typescript 4.x or always use `--legacy-peer-deps`. I appreciate any feedback on this.

**Comment by tangles-0** [comment]
+1

**Comment by timbomckay** [comment]
Man, this comment is from a year ago, I hope you've resolved it by now or moved on to another solution haha. 

If you haven't, I believe you could use NPM's override to force the typescript version.

Something like

```json
"overrides": {
  "twind": {
    "typescript": "$typescript"
  }
}
```

This would tell NPM to use your declared version of Typescript.

I'm not sure if there would be any conflicts, but the install wouldn't complain anymore.
